app/screens/settings-screen/account-screen.tsx

```diff
@@ -139,11 +223,65 @@ export const AccountScreen = ({ navigation }: Props) => {
     })
   }
 
+  const AccountDeletionModal = (
+    <Modal
+      isVisible={modalVisible}
+      onBackdropPress={() => setModalVisible(false)}
+      backdropOpacity={0.3}
+      backdropColor={colors.grey3}
+    >
+      <View style={styles.view}>
+        <Text type="h1">{LL.support.typeDelete({ delete: LL.support.delete() })}</Text>
+        <TextInput style={styles.textInput} onChangeText={setText} value={text} />
+        <GaloyPrimaryButton
+          title="Confirm"
+          disabled={text.toLowerCase().trim() !== LL.support.delete()}
+          onPress={() => {
+            setModalVisible(false)
+            Alert.alert(
+              LL.support.finalConfirmationAccountDeletionTitle(),
+              LL.support.finalConfirmationAccountDeletionMessage(),
+              [
+                { text: LL.common.cancel(), onPress: () => {} },
+                { text: LL.common.ok(), onPress: () => deleteUserAccount() },
+              ],
+            )
+          }}
+          containerStyle={styles.mainButton}
+        />
+        <GaloyTertiaryButton
+          title="Cancel"
+          onPress={() => setModalVisible(false)}
+          outline
+        />
+      </View>
+    </Modal>
+  )
+
   return (
     <Screen preset="scroll">
       {accountSettingsList.map((setting) => (
         <SettingsRow setting={setting} key={setting.id} />
       ))}
+      {AccountDeletionModal}
     </Screen>
   )
 }
+
+const useStyles = makeStyles(({ colors }) => ({
+  view: {
+    marginTop: 50,
```

Comment:
I would get rid of this `marginTop`. It's lowering the modal component and the modal is already partially blocked by the keyboard.